analyzers/src/SonarAnalyzer.CSharp/Rules/InvocationResolvesToOverrideWithParams.cs

```diff
@@ -18,144 +18,106 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-namespace SonarAnalyzer.Rules.CSharp
-{
-    [DiagnosticAnalyzer(LanguageNames.CSharp)]
-    public sealed class InvocationResolvesToOverrideWithParams : SonarDiagnosticAnalyzer
-    {
-        private const string DiagnosticId = "S3220";
-        private const string MessageFormat = "Review this call, which partially matches an overload without 'params'. The partial match is '{0}'.";
-
-        private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
-
-        protected override void Initialize(SonarAnalysisContext context)
-        {
-            context.RegisterNodeAction(
-                c =>
-                {
-                    var invocation = (InvocationExpressionSyntax)c.Node;
-                    CheckCall(c, invocation, invocation.ArgumentList);
-                },
-                SyntaxKind.InvocationExpression);
+namespace SonarAnalyzer.Rules.CSharp;
```

Comment:
All the refactoring in this class is pretty good! There were some ugly cheats that really needed to be moved around, not just syntax-sugared.